src/Features/CSharp/Portable/BraceCompletion/LessAndGreaterThanBraceCompletionService.cs

```diff
@@ -52,7 +52,17 @@ protected override async ValueTask<bool> IsValidOpenBraceTokenAtPositionAsync(Do
         private static async ValueTask<bool> PossibleTypeArgumentAsync(Document document, SyntaxToken token, CancellationToken cancellationToken)
         {
             // type argument can be easily ambiguous with normal < operations
-            if (token.Parent is not BinaryExpressionSyntax node || node.Kind() != SyntaxKind.LessThanExpression || node.OperatorToken != token)
+            if (token.Parent is not BinaryExpressionSyntax(SyntaxKind.LessThanExpression) node || node.OperatorToken != token)
```

Comment:
What magic is this?? Looks like a positional pattern matching with contant inner pattern, but I dont see a deconstruct method for `BinaryExpressionSyntax`

Reply:
NVM, just found [this](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/J/s/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs/SyntaxNodeExtensions.cs,27) Didn't realize it works with extension method too. Neat! :)

Reply:
:D  magic right! i think @alrz added this for us :)

Reply:
Pretty cool indeed. that said, It would make it clearer if go-to-def on the parenthesis can point us to the deconstructor method somehow (if explicitly defined).